### PR TITLE
chore: Deprecate AnchorDefaultProps

### PR DESCRIPTION
**Pull request by fa93hws**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
The type of `defaultProps` will be resolved automatically by `typescript`. There is no need to define our own type.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/Anchor.tsx [hunk 1]
```diff
@@ -70,12 +70,8 @@ export interface AnchorState {
...
 }
 
-export interface AnchorDefaultProps extends AnchorProps {
-  prefixCls: string;
-  affix: boolean;
-  showInkInFixed: boolean;
-  getContainer: () => AnchorContainer;
-}
+// Deprecate
+export type AnchorDefaultProps = AnchorProps;
```
Comment (fa93hws) [inline components/anchor/Anchor.tsx:74]: How to deprecate a type?

File: components/anchor/Anchor.tsx [hunk 2]
```diff
@@ -51,9 +51,9 @@ export interface AnchorProps {
   children?: React.ReactNode;
   offsetTop?: number;
   bounds?: number;
-  affix?: boolean;
-  showInkInFixed?: boolean;
-  getContainer?: () => AnchorContainer;
+  affix: boolean;
+  showInkInFixed: boolean;
+  getContainer: () => AnchorContainer;
```
Comment (afc163) [inline components/anchor/Anchor.tsx:56]: Don't remove `?` that make props required.

Comment (fa93hws) [reply components/anchor/Anchor.tsx:56]: If you've defined the `defaultProps` in the class, typescript is clever enough now to mark them as optional when the component is used. 
image

File: components/anchor/Anchor.tsx (hunk 1 above)
Comment (zombieJ) [reply components/anchor/Anchor.tsx:74]: use jsdoc:
```tsx
/** @deprecated your deprecated reason here ... */
```

Comment (fa93hws) [reply components/anchor/Anchor.tsx:74]: Updated

File: components/anchor/Anchor.tsx (hunk 2 above)
Comment (zombieJ) [reply components/anchor/Anchor.tsx:56]: We can remove `defaultProps` of Anchor. `?` is correct and should not be removed.

**Comment by yoyo837** [comment]
Conflicted

File: components/anchor/Anchor.tsx (hunk 2 above)
Comment (fa93hws) [reply components/anchor/Anchor.tsx:56]: @zombieJ Do you mean that the exported type need to preserve the optional parameter for consumers?

**Comment by afc163** [comment]
Close for no response, feel free to send us a new PR.

**Pull request closed without merging**